lmdeploy/vl/model/llava.py

```diff
@@ -53,18 +53,33 @@ def init_llava_vision_tower(config):
         yield
 
 
+@VISION_MODELS.register_module()
 class LlavaVisionModel(VisonModel):
     """Llava visual model."""
 
-    def __init__(self,
-                 model_path,
-                 arch='LlavaLlamaForCausalLM',
-                 with_llm: bool = False):
+    def __init__(self, model_path, with_llm: bool = False, max_memory=None):
         self.model_path = model_path
-        self.arch = arch
         self.with_llm = with_llm
+        self.max_memory = max_memory
+        from lmdeploy.utils import get_hf_config_content
+        config = get_hf_config_content(model_path)
+        self.arch = config['architectures'][0]
         self.build_model()
 
+    @classmethod
+    def match(cls, config: dict):
+        """check whether the config match the model."""
+        arch = config['architectures'][0]
+        if arch in ['LlavaLlamaForCausalLM', 'LlavaMistralForCausalLM']:
+            mm_vision_tower = config.get('mm_vision_tower', '')
+            projector_type = config.get('mm_projector_type', 'linear')
+            if '_Norm' in projector_type:
```

Comment:
What does those if-statements indicate?

Reply:
Oh, I see. The `Yi` and `internvl_llava` has `_Norm` and `OpenGVLab` settings respectively